### PR TITLE
Duplicate options

### PR DESCRIPTION
**Pull request by chrismcg14**
**Is this a bugfix, or an enhancement? Please describe.**
A clear and concise description of the bug you are fixing (reference issue number), or enhancement you are implementing.

This is a bugfix. When duplicating a job option, most of the properties in the new option did not match that of the original option. 

**Describe the solution you've implemented**

Updated the toMap() Option method to properly set the properties on the returned map. Updated the _setOptionFromParams() function to properly set the properties on the new option based off returned map from the toMap method. Added inputType property to Options class.

### CODE REVIEW

**Comment by chrismcg14** [comment]
Related PRO issue: https://github.com/rundeckpro/rundeckpro/issues/1369

**Review by gschueler — CHANGES_REQUESTED** [review]
this shouldn't avoid adding a new field to the domain class.

step 1 should be create a new test case that fails with the existing behavior, and then proceed from there for implementing the solution. 

I see changes that don't seem related to the problem (updating the _validateOption method), and no test cases showing what that is meant to accomplish.

any changes to the Option or EditOptsController class should have accompanying test cases.  The tests should be the way to answer the question: "what is this supposed to do?" the tests should fail with existing code, and pass with the new code, showing that the behavior was broken and is now fixed.

File: rundeckapp/grails-app/domain/rundeck/Option.groovy [hunk 1]
```diff
@@ -205,12 +208,25 @@ public class Option implements Comparable{
...
+            map.inputType = inputType
+        }
         if(secureInput){
             map.secure=secureInput
         }
+        if(enforced == true) {
+            map.put("enforcedType", "enforced")
+            map.enforced = true
+        }
+        if(enforced == "regex"){
```
Comment (gschueler) [inline rundeckapp/grails-app/domain/rundeck/Option.groovy:221]: this doesn't make sense, the `enforced` field is a Boolean, so can't be compared to a String

**Comment by gschueler** [comment]
I think i found the true source of the bug: the _duplicateOption method passes the value of `option.toMap()` into the `params` when calling _applyOptionAction here http://github.com/rundeck/rundeck/blob/main/rundeckapp/grails-app/controllers/rundeck/controllers/EditOptsController.groovy#L1001-L1001

However, the input "params" map from (form fields sent in the http request) is different than the toMap() map.  normally the requests that modify the options list parse the "params" map into an Option value, however with the Duplicate request, there is not a set of input params, so the toMap() was used instead.

**Comment by gschueler** [comment]
the _applyOptionAction call can instead pass `params: _getParamsFromOption(newOption)`.

**Comment by gschueler** [comment]
different fix https://github.com/rundeck/rundeck/pull/6784

**Pull request closed without merging**